### PR TITLE
v2 API. Close alerts on success

### PR DESCRIPTION
**Pull request by hipyhop**
I made some tweaks to the plugin to automatically close Alerts when the job succeeds.

In the process I noticed that the v1 API endpoint will be completed shutdown in June 2018, and updated the plugin to use the v2 API.

Please take a look and let me know what you think

### CODE REVIEW

**Comment by iksaif** [comment]
Sorry I missed this review ! Looking now

File: src/OpsGenieNotification.groovy [hunk 1]
```diff
@@ -109,27 +110,43 @@ def sendAlert(Map executionData, Map configuration) {
...
 
     // process the response.
-    def response = connection.content.text
-    System.err.println("DEBUG: response: " + response)
-    JsonNode jsnode = json.readTree(response)
-    def status = jsnode.get("status").asText()
-    if (!"success".equals(status)) {
-        System.err.println("ERROR: OpsGenieNotification plugin status: " + status)
+    int response_code = connection.getResponseCode()
+    if (response_code != 202) {
```
Comment (iksaif) [inline src/OpsGenieNotification.groovy:144]: I guess anything with <300 should be valid

**Review by iksaif — APPROVED** [review]

**Comment by iksaif** [comment]
Small comment that can be  fixed in another review, merging this now

File: src/OpsGenieNotification.groovy (hunk 1 above)
Comment (hipyhop) [reply src/OpsGenieNotification.groovy:144]: Possibly but the OpsGenie documentation does say that a 202 will be returned. This is also the same for the Close Alert endpoint.

"Create alert requests are processed asynchronously, therefore valid requests are responded with HTTP status 202 - Accepted."

https://docs.opsgenie.com/docs/alert-api#section-create-alert

**Pull request merged**